### PR TITLE
✨Source Airtable: add table name to data records

### PR DESCRIPTION
**Pull request by askarpets**
## What
Add `_airtable_table_name` field to records structure which is used for returning source table name for the specific stream
https://github.com/airbytehq/airbyte/issues/28177

## How
The table name is taken from the AirtableTables stream and then passed to the corresponding AirtableStream object

### CODE REVIEW

**Comment by airbyte-oss-build-runner** [comment]
##  source-airtable test report (commit [`a8f451b0dc`](https://github.com/airbytehq/airbyte/pull/31044/commits/a8f451b0dcfaf49a7aee8559c69b90763df9b00b)) - ❌

⏲️  Total pipeline duration: 01mn18s 

| Step                                                         | Result   |
|:-------------------------------------------------------------|:---------|
| Connector package install                                    | ✅        |
| Build source-airtable docker image for platform linux/x86_64 | ✅        |
| Unit tests                                                   | ✅        |
| Acceptance tests                                             | ❌        |
| Code format checks                                           | ✅        |
| Validate metadata for source-airtable                        | ✅        |
| Connector version semver check                               | ✅        |
| Connector version increment check                            | ❌        |
| QA checks                                                    | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/28177-source-airtable-add-table-name-to-record/1696414119/a8f451b0dcfaf49a7aee8559c69b90763df9b00b/source-airtable/3.0.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:a8f451b0dcfaf49a7aee8559c69b90763df9b00b)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-airtable test
```

**Review by davydov-d — CHANGES_REQUESTED** [review]
Please update the changelog and bump version in the Dockerfile and metadata file

**Comment by airbyte-oss-build-runner** [comment]
##  source-airtable test report (commit [`71b464c136`](https://github.com/airbytehq/airbyte/pull/31044/commits/71b464c136cb802d9f90e3aee009c4157733c018)) - ❌

⏲️  Total pipeline duration: 02mn19s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Connector package install                                       | ✅        |
| Build source-airtable docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                      | ✅        |
| Acceptance tests                                                | ❌        |
| Code format checks                                              | ✅        |
| Validate metadata for source-airtable                           | ✅        |
| Connector version semver check                                  | ✅        |
| Connector version increment check                               | ✅        |
| QA checks                                                       | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/28177-source-airtable-add-table-name-to-record/1696927068/71b464c136cb802d9f90e3aee009c4157733c018/source-airtable/4.0.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:71b464c136cb802d9f90e3aee009c4157733c018)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-airtable test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-airtable test report (commit [`71b464c136`](https://github.com/airbytehq/airbyte/pull/31044/commits/71b464c136cb802d9f90e3aee009c4157733c018)) - ✅

⏲️  Total pipeline duration: 01mn16s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Connector package install                                       | ✅        |
| Build source-airtable docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                      | ✅        |
| Acceptance tests                                                | ✅        |
| Code format checks                                              | ✅        |
| Validate metadata for source-airtable                           | ✅        |
| Connector version semver check                                  | ✅        |
| Connector version increment check                               | ✅        |
| QA checks                                                       | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/28177-source-airtable-add-table-name-to-record/1696927740/71b464c136cb802d9f90e3aee009c4157733c018/source-airtable/4.0.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:71b464c136cb802d9f90e3aee009c4157733c018)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-airtable test
```

File: airbyte-integrations/connectors/source-airtable/Dockerfile [hunk 1]
```diff
@@ -34,5 +34,5 @@ COPY source_airtable ./source_airtable
 ENV AIRBYTE_ENTRYPOINT "python /airbyte/integration_code/main.py"
 ENTRYPOINT ["python", "/airbyte/integration_code/main.py"]
 
-LABEL io.airbyte.version=4.0.0
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-airtable/Dockerfile:37]: when adding a field or a stream, we should increment minor version, not patch

**Review by davydov-d — APPROVED** [review]

File: airbyte-integrations/connectors/source-airtable/Dockerfile (hunk 1 above)
Comment (askarpets) [reply airbyte-integrations/connectors/source-airtable/Dockerfile:37]: Even if this field for internal use only?

**Review by bazarnov — APPROVED** [review]

**Pull request merged**